utils.go

```diff
@@ -87,12 +87,11 @@ func getRootlessRuntimeDir(rootlessUID int) (string, error) {
 	if err == nil {
 		return runtimeDir, nil
 	}
-	tmpDir := fmt.Sprintf("/run/user/%d", rootlessUID)
+	tmpDir := fmt.Sprintf("%s/storage-user-%d", os.TempDir(), rootlessUID)
```

Comment:
What will happen if your UID is 1000, and another user (UID=2000) has already created `/tmp/storage-user-1000`?
Can 2000 trick to obtain privileges for 1000?

Reply:
Podman should verify that the path has the correct ownership when it is done.  If it is not owned by the current user then it should throw an error

Reply:
TOCTOU?

Reply:
If I check if a file/directory is owned by me and has 0700, then no other non privileged user can change it.  If it is a privileged process screwing around with me, I have far greater problems.

Reply:
It’s still a DOS at the very least.